src/reducers/inputFields.js

```diff
@@ -66,13 +83,28 @@ export default function inputReducer(state = initialState, action) {
         jsonString: ""
       };
     case SET_VALUES:
+      let parseError = false;
+      let parsedValues = { label: "", type: "", options: [] };
+      try {
+        parsedValues = JSON.parse(action.jsonString);
+      } catch (e) {
+        parseError = true;
+      }
+
       return {
         ...state,
-        options: action.setFormValues.hasOwnProperty("options")
-          ? action.setFormValues.options
+        options: parsedValues.hasOwnProperty("options")
+          ? parsedValues.options
           : [],
-        label: action.setFormValues.label,
-        type: action.setFormValues.type
+        label: parsedValues.label,
+        type: parsedValues.type,
+        parsedValues: parsedValues,
+        inputErrorType: !isValidType(parsedValues.type),
```

Comment:
I think it would be better, if we have one array `errors`, where you add the errors and then inside Input.js you just loop over the error